tst/EnergyPlus/unit/HVACDXSystem.unit.cc

```diff
@@ -403,4 +406,167 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_DOASDXCoilTest)
+    ScheduleManager::ProcessScheduleInput();
+    ScheduleManager::Schedule(1).CurrentValue = 1.0; // Enable schedule without calling schedule manager
+
+    int DXSystemNum = 1;
+    bool FirstHVACIteration = true;
+    bool HXUnitOn = false;
+    int InletNode = 1;
+    int ControlNode = 2; // same as outlet node number
+
+    HVACDXSystem::GetDXCoolingSystemInput();
+    EXPECT_EQ(HVACDXSystem::DXCoolingSystem(DXSystemNum).Name, "DX COOLING COIL SYSTEM");
+    EXPECT_FALSE(HVACDXSystem::DXCoolingSystem(DXSystemNum).ISHundredPercentDOASDXCoil);
+    EXPECT_EQ(VariableSpeedCoils::VarSpeedCoil(DXSystemNum).Name, "VS DX COOLING COIL");
+    EXPECT_EQ(2, HVACDXSystem::DXCoolingSystem(DXSystemNum).DXSystemControlNodeNum);
+
+    // set up outdoor environment
+    DataEnvironment::OutDryBulbTemp = 35.0;
+    DataEnvironment::OutHumRat = 0.0196;
+    DataEnvironment::OutBaroPress = 101325.0;
+    DataEnvironment::OutWetBulbTemp = 27.0932;
+
+    // set up inputs to test coil control
+    HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletTemp = 18.0;
+    HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletHumRat = 1.0;
+    DataEnvironment::StdRhoAir = 1.2;
+    DataLoopNode::Node(InletNode).MassFlowRate = 5.66336932 * DataEnvironment::StdRhoAir;
+    DataLoopNode::Node(InletNode).Temp = 24.0;
+    DataLoopNode::Node(InletNode).HumRat = 0.012143698;
+    DataLoopNode::Node(InletNode).Enthalpy = 55029.3778; // conditions at 65 % RH
+    DataLoopNode::Node(ControlNode).TempSetPoint = HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletTemp;
+    Real64 RHControlHumRat = 0.01119276; // humrat at 24C, 60% RH
+    DataLoopNode::Node(ControlNode).HumRatMax = RHControlHumRat;
+
+    // test sensible control
+    HVACDXSystem::ControlDXSystem(DXSystemNum, FirstHVACIteration, HXUnitOn);
+    // system meets temperature set point
+    EXPECT_NEAR(HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletTemp, DataLoopNode::Node(ControlNode).Temp, 0.00001);
+    // system was not told to meet humidity ratio set point (since DesiredOutletHumRat = 1.0)
+    EXPECT_GT(DataLoopNode::Node(ControlNode).HumRat, DataLoopNode::Node(ControlNode).HumRatMax);
+    // sensible load met by compressor speed 3
+    EXPECT_EQ(3, HVACDXSystem::DXCoolingSystem(DXSystemNum).SpeedNum);
+
+    // test latent control
+    HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletHumRat = RHControlHumRat;
+    HVACDXSystem::ControlDXSystem(DXSystemNum, FirstHVACIteration, HXUnitOn);
+
+    // system over cools past temperature set point
+    EXPECT_GT(HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletTemp, DataLoopNode::Node(ControlNode).Temp);
+    // system does meet humidity ratio set point
+    EXPECT_NEAR(DataLoopNode::Node(ControlNode).HumRat, DataLoopNode::Node(ControlNode).HumRatMax, 0.0000001);
+    // latent load needed to increase compressor speed to speed 4
+    EXPECT_EQ(4, HVACDXSystem::DXCoolingSystem(DXSystemNum).SpeedNum);
+}
```

Comment:
OMG this is an awesome unit test.  The idf snippet is just only the pieces needed, the tests are succinct, just ... great!

Reply:
Yeah, tight and right is better than easy. And proves RH control works now!